grype/grypeerr/errors.go

```diff
@@ -0,0 +1,6 @@
+package grypeerr
```

Comment:
Nit: Would it be worth hyphenating this to make it more readable? I know the k8s client has a hyphen in the package name, so should be ok?

Reply:
It's okay for hyphens to appear in module paths, but not package names. If we inserted a hyphen so that we have `package grype-err`, the code wouldn't compile.

[Effective Go](https://golang.org/doc/effective_go.html#package-names) (another great read for aspiring Go devs, btw!) puts forth the convention of packages that use "lower case, single-word names; there should be no need for underscores or mixedCaps". I don't think we need to apply this rule to 100% of cases, but it's worth keeping in mind.

In short, I'm okay with leaving the code as written. If we want to consider alternative approaches, one might be to name this package just "err" or something. So full path would be `github.com/anchore/grype/grype/err`. This would likely lead to aliasing the package when imported throughout our code, for example:

```go
import (
    grypeerr "github.com/anchore/grype/grype/err"
)
```

So I'm not sure it's a net win to change it.

Reply:
I agree with both of you --I don't like `grypeerr` but also agree with Dan's points and don't see a better alternative. The `err` name is problematic since, when imported, would be overridden by the most popular local variable name for errors within functions. I'll leave it as is for now.